addons/account/tests/test_early_payment_discount.py

```diff
@@ -788,3 +788,81 @@ def test_mixed_epd_with_rounding_issue(self):
                 }) for price_unit, quantity, taxes in line_create_vals
             ]
         })
+
+    def test_mixed_epd_with_tax_refund(self):
+        """
+        Ensure epd line are addeed to refunds
+        """
+        self.env.company.early_pay_discount_computation = 'mixed'
+
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'invoice_date': '2022-02-21',
+            'invoice_payment_term_id': self.early_pay_10_percents_10_days.id,
+            'invoice_line_ids': [
+                Command.create({
+                    'price_unit': 100.0,
+                    'quantity': 1,
+                    'tax_ids': [Command.set(self.product_a.taxes_id.ids)],
+                })
+            ]
+        })
+        invoice.action_post()
+
+        move_reversal = self.env['account.move.reversal'].with_context(active_model="account.move", active_ids=invoice.ids).create({
+            'date': fields.Date.from_string('2017-01-01'),
+            'reason': 'no reason again',
+            'refund_method': 'cancel',
+            'journal_id': invoice.journal_id.id,
+        })
+        reversal = move_reversal.reverse_moves()
+        reverse_move = self.env['account.move'].browse(reversal['res_id'])
+
+        receivable_line = invoice.line_ids.filtered(lambda x: x.account_id.account_type == 'asset_receivable')
+        self.assertTrue(receivable_line.full_reconcile_id, "Invoice should be fully reconciled with the credit note")
+
+        self.assertRecordValues(reverse_move.line_ids.sorted('id'), [
+            {
+                'amount_currency': 100.0,
+                'debit': 100.0,
+                'credit': 0.0,
```

Comment:
Helps reducing the size of the test a little bit
```suggestion
                'balance': 100.0,
```